index.js

```diff
@@ -45,13 +48,29 @@ module.exports = function gulpTestCafe (opts) {
 
                 var runner = testcafe.createRunner();
 
-                return runner
+                runner
                     .src(files)
                     .browsers(opts.browsers)
                     .filter(opts.filter)
-                    .screenshots(opts.screenshotsPath, opts.takeScreenshotsOnFail)
-                    .reporter(opts.reporter, opts.reportOutStream)
-                    .run(opts);
+                    .screenshots(opts.screenshotsPath, opts.takeScreenshotsOnFail);
+
+                if (opts.reportOutStream)
+                    opts.reporter.push({ outStream: opts.reportOutStream });
+
+                opts.reporter.forEach(function (reporter) {
+                    var outStream = reporter.outStream;
```

Comment:
I guess it would make the code more clear if you check the type of the `reporter` value at begin, like
```js
if(typeof reporter === 'string')
    runner.reporter(...);
else {
    var name = reporter.name;
    var outStream = ...
}
```

If it's a string you don't need to have other checks.